osquery/core/sha1.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <iomanip>
+#include <sstream>
+
+#include <boost/uuid/sha1.hpp>
+
+#include <osquery/sha1.h>
+
+namespace osquery {
+
+std::string getBufferSHA1(const void* buffer, size_t size) {
+  uint32_t digest[5];
+  boost::uuids::detail::sha1 sha1;
+  sha1.process_bytes(buffer, size);
+  sha1.get_digest(digest);
+
+  // TODO: Determine correct bit manipulation to yield expected SHA1 sum.
```

Comment:
@theopolis, any idea why this step is even necessary (i.e. why the output of Boost's SHA1 implementation doesn't match up with OpenSSL's or sha1sum)?

I'm not very familiar with these hashing algorithms, so I haven't identified anything in the upstream source to clue me in yet. I'll continue reviewing the implementations, but I figured I'd ask in case you already have the experience to point out the problem.